### PR TITLE
Read-DbaXEFile - Refactoring to get the command working again

### LINKED ISSUE #7737: Read-DbaXEFile - Test-Path RemoteTargetFile False Without Wildcard
### Verified issue does not already exist?

Yes

### What error did you receive?

```powershell
PSMessageDetails      : 
Exception             : System.Exception: 
\\ServerName\C$\Program Files\Microsoft SQL Server\MSSQL14.MSSQLSERVER\MSSQL\Log\XeSessionName 
                        cannot be accessed from LocalServerName. Does domain\user have access?
TargetObject          : 
CategoryInfo          : NotSpecified: (:) [], Exception
FullyQualifiedErrorId : dbatools_Read-DbaXEFile
ErrorDetails          : 
InvocationInfo        : System.Management.Automation.InvocationInfo
ScriptStackTrace      : at Stop-Function, 
                 C:\Program Files\WindowsPowerShell\Modules\dbatools\1.1.11\allcommands.ps1: line 91469
                        at Read-DbaXEFile<Process>, 
                 C:\Program Files\WindowsPowerShell\Modules\dbatools\1.1.11\allcommands.ps1: line 59945
                        at Select-DefaultView<Process>, 
                 C:\Program Files\WindowsPowerShell\Modules\dbatools\1.1.11\allcommands.ps1: line 88935
                        at Get-DbaXESession<Process>, 
                 C:\Program Files\WindowsPowerShell\Modules\dbatools\1.1.11\allcommands.ps1: line 39354
                        at <ScriptBlock>, <No file>: line 1
PipelineIterationInfo : {}
```

### Steps to Reproduce

```powershell
Get-DbaXESession $SqlInstance -Session $XeSessionName -EnableException | Read-DbaXEFile -EnableException
```

### Are you running the latest release?

Yes

### Other details or mentions

I *think* the issue was introduced in https://github.com/sqlcollaborative/dbatools/commit/a69925a68f03e26918ff7b3628f832b481cb090c# with the removal of [lines 101 to 109](https://github.com/sqlcollaborative/dbatools/commit/a69925a68f03e26918ff7b3628f832b481cb090c#diff-12744f7e37b1d7ab4e5229ae59bc3da938289bb5769b70317c2fb2414d3b41b9L101-L109). 

Now the below line would return `$false` on all `(Get-DbaXESession).RemoteTargetFile`s.

https://github.com/sqlcollaborative/dbatools/blob/fb2f08f69524af3f85fb0b6617a3b1d3419a41cd/functions/Read-DbaXEFile.ps1#L90

Considering the aforementioned commit, I think the the other `Read*` cmdlets may be impacted.

### What PowerShell host was used when producing this error

PowerShell Core (pwsh.exe), Windows PowerShell (powershell.exe), VS Code (integrated terminal)

### PowerShell Host Version

```pwsh

Name                           Value
----                           -----
PSVersion                      7.1.4
PSEdition                      Core
GitCommitId                    7.1.4
OS                             Microsoft Windows 10.0.18363
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

```powershell
Name                           Value     
----                           -----                               
PSVersion                      5.1.18362.1171         
PSEdition                      Desktop                
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
BuildVersion                   10.0.18362.1171        
CLRVersion                     4.0.30319.42000        
WSManStackVersion              3.0                    
PSRemotingProtocolVersion      2.3                    
SerializationVersion           1.1.0.1                
```

### SQL Server Edition and Build number

```sql
Microsoft SQL Server 2017 (RTM-CU24) (KB5001228) - 14.0.3391.2 (X64) 
Apr 28 2021 10:32:18 
Copyright (C) 2017 Microsoft Corporation
Enterprise Edition: Core-based Licensing (64-bit) 
on Windows Server 2016 Standard 10.0 <X64> (Build 14393: ) (Hypervisor)
```

### PR DESCRIPTION
**Pull request by andreasjordan**
## Please read -- recent changes to our repo
On November 10, 2022, [we removed some bloat from our repository (for the second and final time)](https://github.com/dataplat/dbatools/issues/8542). This change requires that all contributors reclone or refork their repo.

PRs from repos that have not been recently reforked or recloned will be closed and @potatoqualitee will cherry-pick your commits and open a new PR with your changes.

 - [ ] Please confirm you have the smaller repo (85MB .git directory vs 275MB or 110MB or 185MB .git directory)

## Type of Change

 - [x] Bug fix (non-breaking change, fixes #7737 )
 - [ ] New feature (non-breaking change, adds functionality, fixes # )
 - [ ] Breaking change (affects multiple commands or functionality, fixes # )
 - [ ] Ran manual Pester test and has passed (`.\tests\manual.pester.ps1`)
 - [ ] Adding code coverage to existing functionality
 - [ ] Pester test is included
 - [ ] If new file reference added for test, has is been added to github.com/dataplat/appveyor-lab ?
 - [ ] Unit test is included
 - [ ] Documentation
 - [ ] Build system

See issue for details.

### CODE REVIEW

File: public/Read-DbaXEFile.ps1 [hunk 1]
```diff
@@ -54,83 +54,80 @@ function Read-DbaXEFile {
...
+                Write-Message -Level Verbose -Message "Received $($files.Count) files based on [$targetFile]"
+            } else {
+                Stop-Function -Message "The Path [$pathObject] has an unsupported file type of [$($pathObject.GetType().FullName)]."
             }
 
-            $accessible = Test-Path -Path $currentfile
-            $whoami = whoami
+            foreach ($file in $files) {
+                $accessible = Test-Path -Path $file
+                $whoami = whoami
```
[the remark this answers is not in the document]
Comment (andreasjordan) [reply public/Read-DbaXEFile.ps1:87]: I guess not. Maybe we only test if Path is a filename.

[the remark this answers is not in the document]
Comment (andreasjordan) [reply public/Read-DbaXEFile.ps1:87]: Problem anyway: Get-ChildItem will just silently retun no files if the path has a '*' wildcard.

File: public/Read-DbaXEFile.ps1 [hunk 2]
```diff
@@ -54,83 +54,80 @@ function Read-DbaXEFile {
...
-            }
+                # use the SqlServer\Read-SqlXEvent cmdlet from Microsoft
+                # because the underlying Class uses Tasks
+                # which is hard to handle in PowerShell
 
-            # use the SqlServer.XEvent\Read-SqlXEvent cmdlet from Microsoft
-            # because the underlying Class uses Tasks
-            # which is hard to handle in PowerShell
+                if ($Raw) {
+                    SqlServer\Read-SqlXEvent -FileName $file
```
Comment (wsmelton) [inline public/Read-DbaXEFile.ps1:99]: Not fond that we tell everyone that we know dbatools v2 does not play nice with sqlserver module when they are in the same session and turn around and have it in a command. What affect is had if someone uses this command, sqlserver imports, and then try using commands that do not play well?

This is missing all kinds of validation that (1) the module is installed and (2) error handling on the calling the command. (There is not one try/catch in this command).

File: public/Read-DbaXEFile.ps1 [hunk 3]
```diff
@@ -54,83 +54,80 @@ function Read-DbaXEFile {
...
+            } elseif ($pathObject -is [System.IO.FileInfo]) {
+                $files = $pathObject.FullName
+            } elseif ($pathObject -is [Microsoft.SqlServer.Management.XEvent.Session]) {
+                if ($pathObject.TargetFile.Length -eq 0) {
+                    Stop-Function -Message "The session [$pathObject] does not have an associated Target File." -Continue
                 }
 
-                $instance = [dbainstance]$file.ComputerName
-
+                $instance = [dbainstance]$pathObject.ComputerName
```
Comment (wsmelton) [inline public/Read-DbaXEFile.ps1:71]:
```suggestion
                $instance = [DbaInstance]$pathObject.ComputerName
```

File: public/Read-DbaXEFile.ps1 (hunk 1 above)
[the remark this answers is not in the document]
Comment (wsmelton) [reply public/Read-DbaXEFile.ps1:87]: I'm referring to the whoami call

**Comment by andreasjordan** [comment]
I know that this command is not perfect. Shawn has some valid points.

But: Currently, the command is not working at all. So everything that works in some environments is better. I would like to get this command going to be able for folks to test it and see how it can be used. Then we can shape it.

I don't like to invest a lot of hours into a command that then is removed from the module. I hope you can comprehend that.

File: public/Read-DbaXEFile.ps1 (hunk 1 above)
[the remark this answers is not in the document]
Comment (andreasjordan) [reply public/Read-DbaXEFile.ps1:87]: Do we really need to tell the user which account is used? Maybe remove it...

**Comment by andreasjordan** [comment]
Added some more changes including documentation and error handling.

**Comment by andreasjordan** [comment]
Now using Read-XEvent from dbatools.library, so SqlServer module not needed.

**Comment by potatoqualitee** [comment]
🔥 🔥

**Pull request merged**